flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/transform/PreTransformProcessor.java

```diff
@@ -62,30 +62,18 @@ public CreateTableEvent preTransformCreateTableEvent(CreateTableEvent createTabl
     public BinaryRecordData processFillDataField(BinaryRecordData data) {
         List<Object> valueList = new ArrayList<>();
         List<Column> columns = tableChangeInfo.getPreTransformedSchema().getColumns();
-        for (int i = 0; i < columns.size(); i++) {
-            valueList.add(
-                    getValueFromBinaryRecordData(
-                            columns.get(i).getName(),
-                            data,
-                            tableChangeInfo.getSourceSchema().getColumns(),
-                            tableChangeInfo.getSourceFieldGetters()));
+        Map<String, RecordData.FieldGetter> sourceFieldGetters =
+                tableChangeInfo.getSourceFieldGetters();
+        for (Column column : columns) {
+            RecordData.FieldGetter fieldGetter = sourceFieldGetters.get(column.getName());
+            if (fieldGetter != null) {
```

Comment:
I think `TableChangeInfo` should be able to guarantee the invariability that for each column in `preTransformedSchema`, there must be corresponding `fieldGetters` available.

Reply:
done

Reply:
@yuxiqian PTAL~